keras_nlp/models/deberta/deberta_encoder.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from tensorflow import keras
+
+from keras_nlp.models.deberta.disentangled_self_attention import (
+    DisentangledSelfAttention,
+)
+from keras_nlp.utils.keras_utils import clone_initializer
+
+from keras_nlp.layers.transformer_layer_utils import (  # isort:skip
+    merge_padding_and_attention_mask,
+)
+
+
+class DebertaEncoder(keras.layers.Layer):
```

Comment:
This definitely feels like a lot of duplication, where we are mirroring a ton of arguments just for the shared embedding weights.

I wonder if instead we could have a single `Deberta` function model and remove this class, and instead build a shared `RelativeEmbeddings` layer.

Init signature `RelativeEmbeddings(bucket_size, hidden_dim, kernel_initializer)`. In model call, it would do something like the following:

```
def call(inputs):
    sequence_length = tf.shape(inputs)[-2]
    rel_attn_span = self.bucket_size
    x = self.rel_embeddings[
        self.bucket_size - rel_attn_span : self.bucket_size + rel_attn_span,
        :,
    ]
    # Potentially could return with batch_size shape on first dim too??
    return x[tf.newaxis, :sequence_length]
```

This layer would be created along with our other main "deberta" layers, and passed all the way to `DisentangledSelfAttention` layer during init, and saved a member (`self.relative_embedding = relative_embedding`). The layer would be called on the query input tensor inside `DisentangledSelfAttention`. That would save a lot of the duplication here, and still allow sharing the weights where we need (by writing a single layer to encapsulate the shared weights).

Reply:
Oops meant to leave this comment on `DebertaCustomEncoder`!